### ISSUE TITLE
Custom error views options are broken

### ISSUE THREAD

**Issue by mhsdesign** [neos/flow-development-collection#2636]
Currently, custom error views https://flowframework.readthedocs.io/en/latest/TheDefinitiveGuide/PartIII/ErrorAndExceptionHandling.html#custom-error-views can be defined like this (as shown in the docs):

[link](https://github.com/neos/flow-development-collection/blob/0fd339c365397ed27430b1ce47963ec7761cb934/Neos.Flow/Configuration/Settings.Error.yaml#L33)
```yaml
Neos:
  Flow:
    error:
      exceptionHandler:
        renderingGroups:
          databaseConnectionExceptions:
            matchingExceptionClassNames: ['Neos\Flow\Persistence\Doctrine\Exception\DatabaseException']
            options:
              templatePathAndFilename: 'resource://Neos.Flow/Private/Templates/Error/Default.html'
              variables:
                errorDescription: 'Sorry, the database connection couldn''t be established.'
```

## Problem Nr 1.
the one problem is, that the template 'resource://Neos.Flow/Private/Templates/Error/Default.html' is fluid and flow is now independent of fluid and uses the Neos\Flow\Mvc\View\SimpleTemplateView [link](https://github.com/neos/flow-development-collection/blob/0fd339c365397ed27430b1ce47963ec7761cb934/Neos.Flow/Configuration/Settings.Error.yaml#L16) for template resolution unless the fluid adaptor is installed, which will override it and use: Neos\FluidAdaptor\View\StandaloneView [link](https://github.com/neos/flow-development-collection/blob/a8f81b4f3ddafe30f50dcd6092f98cb3c651004d/Neos.FluidAdaptor/Configuration/Settings.yaml#L19)

## Problem Nr 2.
and the second problem is that using `options.templatePathAndFilename` or really anything like `options.(templatePathAndFilename|layoutRootPath|partialRootPath|format)` are not set with `viewClassName->setOptions($options)` but are set with ObjectAccess::setProperty directly on the view instance. [link](https://github.com/neos/flow-development-collection/blob/0fd339c365397ed27430b1ce47963ec7761cb934/Neos.Flow/Classes/Error/AbstractExceptionHandler.php#L184)

That means that the view needs to work with object properties like `(templatePathAndFilename|layoutRootPath|partialRootPath|format)` but that's not the case for the default SimpleTemplateView and not the Neos way to do this. One is supposed to set the options with $view->setOption.
It looks the object access like this was done and is possible only in favor of the original (the extended adapted) fluid view from Typo3: https://github.com/TYPO3/Fluid/blob/a60a9219ab556104c70abb3efad45caa2c188aa5/src/View/TemplatePaths.php#L93
This view has the above and in the docs mentioned object properties and can work with them (if they are set directly).

## Solution:
remove/deprecate `options.(templatePathAndFilename|layoutRootPath|partialRootPath|format)` and let these options only be controlled with the undocumented but working: `options.viewOptions` which will be set with $view->setOption(). The only problem is that the current object options don't map in case of the fluid view directly to the setOption alternative provided by the adaptor.

###  This 'legacy' way:
```yaml
options:
  templatePathAndFilename: 'file'
  layoutRootPath: 'path'
  partialRootPath: 'path'
  format: 'html'
```

### can (nearly*) be translated yet already to this:

there is no supported Option for: format: 'html' [link](https://github.com/neos/flow-development-collection/blob/cf625fb60dab77b9e5962ce7e75e4c1113c8f61e/Neos.FluidAdaptor/Classes/View/AbstractTemplateView.php#L36)
```
options:
  viewOptions:
    templatePathAndFilename: 'file'
    layoutRootPaths: ['path']
    partialRootPaths: ['path']
```

i say nearly because we arrived at:

## Problem Nr 3.
in short due to stuff ^^ the above alternative will not work yet as there is a if statement in both of the exception handlers to first check if `options.templatePathAndFilename` passes the `isset()` check [link 1](https://github.com/neos/flow-development-collection/blob/6299f54cd3a6dd3edd78c5b4fe601d49b95ba107/Neos.Flow/Classes/Error/DebugExceptionHandler.php#L67) [link 2](https://github.com/neos/flow-development-collection/blob/8c5d05784eb2113460ee4096abe174581aa8f916/Neos.Flow/Classes/Error/ProductionExceptionHandler.php#L40) but that can be removed. see issue #1108

**Comment by mhsdesign** [comment]
maybe we could deprecate the use of `options.(templatePathAndFilename|layoutRootPath|partialRootPath|format)`
with Neos 7.3?

i would take care of adjusting the documentation (including to document the options.viewOptions and options.viewClassName)

and with the 'breaking' Neos 8 the legacy way could be kicked and the yaml exceptionHandler configs from Neos and Flow be adjusted and Problem 3 solved.

**Comment by kdambekalns** [comment]
Hehe… I win this race: https://github.com/neos/flow-development-collection/issues/1108 😎

**Comment by mhsdesign** [comment]
hmm youre right 🤔

**Comment by mhsdesign** [comment]
i though i found more infos but actually its quite similar ... should i make a comment out of this and close this?

**Comment by sorenmalling** [comment]
> i though i found more infos but actually its quite similar ... should i make a comment out of this and close this?

@mhsdesign Your comments in #1108 are great 👍  I will close this issue :)
